drivers/sensor/max31875/max31875.c

```diff
@@ -0,0 +1,266 @@
+			LOG_DBG("Failed to set attribute!");
+			return -EIO;
+		}
+		break;
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		/* conversion rate in mHz */
+		cr = val->val1 * 1000 + val->val2 / 1000;
+
+		/* the sensor supports 0.25Hz, 1Hz, 4Hz and 8Hz */
+		/* conversion rate */
+		switch (cr) {
+		case 250:
+			value = MAX31875_CONV_PER_SEC_0_25 << MAX31875_CONV_PER_SEC_SHIFT;
+			break;
+
+		case 1000:
+			value = MAX31875_CONV_PER_SEC_1 << MAX31875_CONV_PER_SEC_SHIFT;
+			break;
+
+		case 4000:
+			value = MAX31875_CONV_PER_SEC_4 << MAX31875_CONV_PER_SEC_SHIFT;
+			break;
+
+		case 8000:
+			value = MAX31875_CONV_PER_SEC_8 << MAX31875_CONV_PER_SEC_SHIFT;
+			break;
+
+		default:
+			return -ENOTSUP;
+		}
+
+		if (max31875_update_config(dev, MAX31875_CONV_PER_SEC_MASK, value) < 0) {
+			LOG_DBG("Failed to set attribute!");
+			return -EIO;
+		}
+
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int max31875_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	struct max31875_data *data = dev->data;
+	const struct max31875_config *cfg = dev->config;
+	uint16_t val;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_AMBIENT_TEMP);
+
+	if (max31875_reg_read(cfg, MAX31875_REG_TEMPERATURE, &val) < 0) {
+		return -EIO;
+	}
+
+	if (data->config_reg & (1 << MAX31875_DATA_FORMAT_SHIFT)) {
+		data->sample = arithmetic_shift_right((int16_t)val,
+							  MAX31875_DATA_FORMAT_EXTENDED_SHIFT);
+	} else {
+		data->sample = arithmetic_shift_right((int16_t)val,
+							  MAX31875_DATA_FORMAT_NORMAL_SHIFT);
+	}
+
+	return 0;
+}
+
+static int max31875_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct max31875_data *data = dev->data;
+	int32_t uval;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	uval = data->sample * MAX31875_TEMP_SCALE;
+	val->val1 = uval / 1000000;
+	val->val2 = uval % 1000000;
+
+	return 0;
+}
+
+static const struct sensor_driver_api max31875_driver_api = {
+	.attr_set = max31875_attr_set,
+	.sample_fetch = max31875_sample_fetch,
+	.channel_get = max31875_channel_get,
+};
+
+int max31875_init(const struct device *dev)
+{
+	const struct max31875_config *cfg = dev->config;
+	struct max31875_data *data = dev->data;
+
+	if (!device_is_ready(cfg->bus.bus)) {
+		LOG_ERR("I2C dev %s not ready", cfg->bus.bus->name);
+		return -EINVAL;
+	}
+
+	data->config_reg = MAX31875_CONFIG(cfg->data_format, cfg->resolution,
+		cfg->conversions_per_second);
+
+	return max31875_update_config(dev, 0, 0);
+}
+
+
+#define MAX31875_INST(inst) \
+	static struct max31875_data max31875_data_##inst; \
+	static const struct max31875_config max31875_config_##inst = { \
+		.bus = I2C_DT_SPEC_INST_GET(inst), \
+		.conversions_per_second = DT_ENUM_IDX(DT_DRV_INST(inst), conversions_per_second), \
+		.resolution = DT_ENUM_IDX(DT_DRV_INST(inst), resolution), \
```

Comment:
`DT_INST_ENUM_IDX`

Reply:
good one!

Reply:
actually the build fails with `DT_INST_ENUM_IDX`

Reply:
or I just need to replace with `DT_INST_ENUM_IDX(inst, resolution)`